server/AuthController.js

```diff
@@ -12,6 +13,42 @@ module.exports = {
     res.set(JWT.renewTokenHeader, token);
     res.status(200).send({ token })
   },
+  impersonate: (_grabUser, id='id') => async (req, res) => {
+    try {
+      let token = JWT.issue({
+        // The new user will be whatever the target id is
+        user: req.param(id),
+        // Add the previous user to the back of the stack of users ids
+        // Makes it easy to use with lodash :D
+        impersonateStack: _.concat(req.user.id)(req.tokenPayload.impersonateStack || [])
+      })
+      res.set(JWT.renewTokenHeader, token)
+      res.status(200).send({token})
+    } catch (e) {
+      // Return 500 because there was a problem with the server
+      // Either grabbing permissions failed (developer's fault)
+      // Or generating a token failed (server's fault)
+      res.status(500).send({message: e.message || e})
```

Comment:
If you're going to do this, let's use `sails-async` so that this logic isn't duplicated